apps/communications/contacts/views/details/js/details_controller.js

```diff
@@ -0,0 +1,157 @@
+      contact.category.push('favorite');
+    } else {
+      if (!contact.category) {
+        return;
+      }
+      var pos = contact.category.indexOf('favorite');
+      if (pos > -1) {
+        contact.category.splice(pos, 1);
+      }
+    }
+
+    function onError(e) {
+      console.error('Error saving favorite');
+      // We must restore contact in order to update the UX accordingly
+      if (favorite) {
+        var pos = contact.category.indexOf('favorite');
+        if (pos > -1) {
+          contact.category.splice(pos, 1);
+        }
+      } else {
+        contact.category = contact.category || [];
+        contact.category.push('favorite');
+      }
+
+      dispatchEvent('toggleFavoriteDone', {contact: contact});
+    }
+
+    ContactsService.save(
+      utils.misc.toMozContact(contact),
+      function(e) {
+        if (typeof e !== 'undefined') {
+          onError(e);
+          return;
+        }
+
+        ContactsService.get(contact.id, function(savedContact) {
+          dispatchEvent('toggleFavoriteDone', {contact: savedContact});
+        }, onError);
+      }
+    );
+  }
+
+  function init(activity) {
+    if (typeof activity !== 'undefined') {
+      setActivity(activity);
+    }
+
+    window.addEventListener('backAction', handleBackAction);
+    window.addEventListener('editAction', handleEditAction);
+    window.addEventListener('toggleFavoriteAction', toggleFavorite);
+    window.addEventListener('shareAction', shareContact);
+    window.addEventListener('findDuplicatesAction', findDuplicates);
+
+    navigator.mozContacts.oncontactchange = evt => {
+      var json = {
+        contactID: evt.contactID,
+        reason: evt.reason
+      };
+      sessionStorage.setItem('oncontactchange', JSON.stringify(json));
+    };
+  }
+
+  function shareContact(evt) {
+    const VCARD_DEPS = [
+      '/shared/js/text_normalizer.js',
+      '/shared/js/contact2vcard.js',
+      '/shared/js/setImmediate.js'
+    ];
+
+    if (!evt.detail || typeof evt.detail.contact === 'undefined') {
+      console.error('Missing parameters in CustomEvent');
+      return;
+    }
+
+    var contact = evt.detail.contact;
+
+    LazyLoader.load(VCARD_DEPS,function vcardLoaded() {
+      ContactToVcardBlob([contact], function blobReady(vcardBlob) {
+        var filename = VcardFilename(contact);
+         /* jshint nonew: false */
+        new MozActivity({
+          name: 'share',
+          data: {
+            type: 'text/vcard',
+            number: 1,
+            blobs: [new window.File([vcardBlob], filename)],
+            filenames: [filename]
+          }
+        });
+        // The MIME of the blob should be this for some MMS gateways
+      }, { type: 'text/x-vcard'} );
+    });
+  }
+
+  function handleBackAction(evt) {
+    if (_activity) {
+      _activity.postResult({});
+    } else {
+      window.history.back();
+    }
+  }
+
+  function handleEditAction(evt) {
+    // TODO
```

Comment:
Add bug number! https://bugzilla.mozilla.org/show_bug.cgi?id=1169579 ;) 

It would be great if we explain here that in the future we will change the URL to navigate to #update.